modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/util/QueryChecker.java

```diff
@@ -377,7 +391,23 @@ public void check() {
             assertThat("Column types don't match", colTypes, equalTo(expectedColumnTypes));
         }
 
-        var res = CursorUtils.getAllFromCursor(cur);
+        if (metadataMatchers != null) {
+            List<ColumnMetadata> columnMetadata = cur.metadata().columns();
+
+            Iterator<ColumnMetadata> valueIterator = columnMetadata.iterator();
+            Iterator<MetadataMatcher> matcherIterator = metadataMatchers.iterator();
+
+            while (matcherIterator.hasNext() && valueIterator.hasNext()) {
+                MetadataMatcher matcher = matcherIterator.next();
+                ColumnMetadata actualElement = valueIterator.next();
+
+                matcher.check(actualElement);
+            }
+
+            assertFalse(matcherIterator.hasNext(), "Column metadata doesn't match");
```

Comment:
`valueIterator.hasNext()` may be true, which also indicates a mismatch.
Let's replace this with `assertEquals(columnMetadata.size(), metadataMatchers.size())`.

Reply:
ok